src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,98 @@
 package core.basesyntax;
 
+import java.util.Arrays;
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_CAPACITY = 10;
+    private static final float GROW_SIZE = 1.5F;
+    private int size;
+    private T[] arrayList;
+
+    public ArrayList() {
+        arrayList = (T[]) new Object[DEFAULT_CAPACITY];
+    }
+
     @Override
     public void add(T value) {
-
+        if (size == arrayList.length) {
+            resize();
+        }
+        arrayList[size++] = value;
     }
 
     @Override
     public void add(T value, int index) {
-
+        if (index > size || index < 0) {
+            throw new ArrayListIndexOutOfBoundsException("Invalid arrayList index");
+        } else if (size == arrayList.length) {
```

Comment:
it is better to move the logic into a separate method (line 18)

Reply:
Small condition which is executed one time, don't see real necessity to make separate method for it

Reply:
it is executed two times

Reply:
im about line where called resize() method